lib/vquic/curl_ngtcp2.c

```diff
@@ -526,7 +528,13 @@ static CURLcode quic_init_ssl(struct Curl_cfilter *cf,
     SSL_set_alpn_protos(ctx->ssl, alpn, (int)alpnlen);
 
   /* set SNI */
-  SSL_set_tlsext_host_name(ctx->ssl, cf->conn->host.name);
+  if((0 == Curl_inet_pton(AF_INET, cf->conn->host.name, checkip))
+#ifdef ENABLE_IPV6
+     && (0 == Curl_inet_pton(AF_INET6, cf->conn->host.name, checkip))
+#endif
+    ) {
+    SSL_set_tlsext_host_name(ctx->ssl, cf->conn->host.name);
+  }
```

Comment:
I think it should do the SNI lowercase dance the way we do it for regular OpenSSL:

https://github.com/curl/curl/blob/39c883560a3b9ac7ddbec78f7a6a8cc13930f219/lib/vtls/openssl.c#L3765-L3769

Reply:
done